tests/integration/suite/daprd/resiliency/apps/retrymatchinggrpc.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apps
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	commonv1 "github.com/dapr/dapr/pkg/proto/common/v1"
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/grpc/app"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(retrymatchinggrpc))
+}
+
+type retrymatchinggrpc struct {
+	daprd1   *daprd.Daprd
+	daprd2   *daprd.Daprd
+	counters sync.Map
+}
+
+func (d *retrymatchinggrpc) Setup(t *testing.T) []framework.Option {
+	onInvoke := func(ctx context.Context, in *commonv1.InvokeRequest) (*commonv1.InvokeResponse, error) {
+		key := in.GetMethod()
+		if key == "" {
+			return nil, status.Error(codes.InvalidArgument, "key is empty")
+		}
+
+		dataStr := string(in.GetData().GetValue())
+		responseStatusCode, err := strconv.Atoi(dataStr)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+
+		c, _ := d.counters.LoadOrStore(key, &atomic.Int32{})
+		counter := c.(*atomic.Int32)
+		counter.Add(1)
+
+		return nil, status.Error(codes.Code(responseStatusCode), "error for key: "+key)
+	}
+
+	resiliency := `
+apiVersion: dapr.io/v1alpha1
+kind: Resiliency
+metadata:
+  name: myresiliency
+spec:
+  policies:
+    retries:
+      DefaultAppRetryPolicy:
+        policy: constant
+        duration: 10ms
+        maxRetries: 3
+        matching:
+          gRPCStatusCodes: 1,2,5-7
+          httpStatusCodes: 500-599
```

Comment:
Please, can we have a seperate integration test for each of the test matrix. I'd suggest putting this into a `app/retry` package.

We need behaviour for- (HTTP & gRPC): no code, 1 code, multiple code, range code, singular code & range.

Reply:
Done